src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionAnalysis.cs

```diff
@@ -1168,9 +1325,19 @@ public bool RunAnalysis()
             {
                 _rawActions = new Dictionary<string, RGGameAction>();
                 _rawActionsByNode = new Dictionary<SyntaxNode, List<RGGameAction>>();
+                _unboundActions = new Dictionary<string, Dictionary<string, RGGameAction>>();
                 Warnings = new List<RGActionAnalysisWarning>();
-            
-                RunCodeAnalysis();
+
+                {
+                    int passNum = 1;
+                    do
+                    {
+                        _changed = false;
+                        RunCodeAnalysis(passNum);
+                        ++passNum;
+                    } while (_changed);
```

Comment:
Should we also set a max number of passes? Or is it guaranteed to terminate?

Reply:
All of the propagation logic is guaranteed to terminate. Even if we somehow end up with a bug where the analysis never terminates, the user can just click the "Cancel" button on the progress bar and the analysis will abort (I've made sure that all parts of the analysis procedure are interruptible by the user whenever a call to `NotifyProgress` is made).